ooni/common/txextra.py

```diff
@@ -191,12 +197,22 @@ def _handleRedirect(self, response, method, uri, headers, redirectCount):
             response.request.absoluteURI,
             locationHeaders[0]
         )
+        uri = client.URI.fromBytes(location)
+        if self.ignorePrivateRedirects and (is_private_address(uri.host) or
+                                            uri.host == "localhost"):
+            return response
```

Comment:
We've discussed this matter and decided that redirect to private address space is OK and only redirect-to-directly-connected address (including `127.0.0.0/8`) should be avoided. Also, it does not provide *security* as domain name [like](http://www.tp-link.com/us/faq-1170.html) `tplinkwifi.net` can be used in `Location` as well.

IIRC, we've seen blockpages served from a server located in a private address space, so losing ability to see those pages is sort of bad. I'll verify that.